src/main/java/com/box/sdk/BoxSignRequest.java

```diff
@@ -0,0 +1,812 @@
+
+        JsonArray signersJSON = new JsonArray();
+        for (BoxSignRequestSigner signer : signers) {
+            signersJSON.add(signer.getJSONObject());
+        }
+        requestJSON.add("signers", signersJSON);
+
+        JsonArray sourceFilesJSON = new JsonArray();
+        for (BoxSignRequestFile sourceFile : sourceFiles) {
+            sourceFilesJSON.add(sourceFile.getJSONObject());
+        }
+        requestJSON.add("source_files", sourceFilesJSON);
+
+        JsonObject parentFolderJSON = new JsonObject();
+        parentFolderJSON.add("id", parentFolderId);
+        parentFolderJSON.add("type", "folder");
+        requestJSON.add("parent_folder", parentFolderJSON);
+
+        if (optionalParams != null) {
+            optionalParams.appendParamsAsJson(requestJSON);
+        }
+
+        URL url = SIGN_REQUESTS_URL_TEMPLATE.build(api.getBaseURL());
+        BoxJSONRequest request = new BoxJSONRequest(api, url, "POST");
+        request.setBody(requestJSON.toString());
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+        BoxSignRequest signRequest = new BoxSignRequest(api, responseJSON.get("id").asString());
+        return signRequest.new Info(responseJSON);
+    }
+
+    /**
+     * Returns information about this sign request.
+     *
+     * @param fields the fields to retrieve.
+     * @return information about this sign request.
+     */
+    public BoxSignRequest.Info getInfo(String... fields) {
+        QueryStringBuilder builder = new QueryStringBuilder();
+        if (fields.length > 0) {
+            builder.appendParam("fields", fields);
+        }
+        URL url = SIGN_REQUEST_URL_TEMPLATE.buildAlphaWithQuery(
+                this.getAPI().getBaseURL(), builder.toString(), this.getID());
+        BoxAPIRequest request = new BoxAPIRequest(this.getAPI(), url, "GET");
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+        return new BoxSignRequest.Info(responseJSON);
+    }
+
+    /**
+     * Returns all the sign requests.
+     *
+     * @param api    the API connection to be used by the resource.
+     * @param fields the fields to retrieve.
+     * @return an iterable with all the sign requests.
+     */
+    public static Iterable<BoxSignRequest.Info> getAll(final BoxAPIConnection api, String... fields) {
+        return getAll(DEFAULT_LIMIT, api, fields);
+    }
+
+    /**
+     * Returns all the sign requests.
+     *
+     * @param limit  the limit of items per single response. The default value is 100.
+     * @param api    the API connection to be used by the resource.
+     * @param fields the fields to retrieve.
+     * @return an iterable with all the sign requests.
+     */
+    public static Iterable<BoxSignRequest.Info> getAll(int limit, final BoxAPIConnection api, String... fields) {
```

Comment:
Shall we keep `BoxAPIConnection api` as the first argument like in other static methods?